client/src/components/Upload.vue

```diff
@@ -173,13 +181,13 @@ export default {
       const name = files.length === 1 ? files[0].name : '';
       this.preUploadErrorMessage = null;
       try {
-        this.addPendingUpload(name, files);
+        this.addPendingUpload(name, files, this.filetypes);
       } catch (err) {
         this.preUploadErrorMessage = err;
       }
     },
-    addPendingUpload(name, allFiles) {
-      const { type, media, csv } = prepareFiles(allFiles);
+    addPendingUpload(name, allFiles, filetypes) {
```

Comment:
No need to make this a method param.
```suggestion
    addPendingUpload(name, allFiles) {
      // pull regexps from store, or use `mapGetters`
      const { regexps } = this.$store.getters.Filetypes;
```

In general, if you find yourself passing instance properties to instance methods, that's a smell, since they share the same context.